gui/WorldView.cxx

```diff
@@ -441,6 +473,126 @@ void WorldView::setBackgroundColor(QColor const& color)
+  {
+    std::string filename = vtiPath.toStdString();
+
+    vtkNew<vtkXMLImageDataReader> readerIm;
+
+    readerIm->SetFileName(filename.c_str());
+    readerIm->Update();
+
+    int width = readerIm->GetOutput()->GetDimensions()[0];
+    int height = readerIm->GetOutput()->GetDimensions()[1];
+
+    vtkNew<vtkPoints> points;
+
+    points->SetNumberOfPoints(width*height);
+
+    //Unprojecting the points from the vtifile
+
+    double dmToImageRatio = camera->GetImageDimensions()[0]/width;
+
+    vtkMaptkCamera* scaledCam = camera->scaledK(dmToImageRatio);
+
+    double pixel[3];
+
+    for (vtkIdType idPixel = 0; idPixel < readerIm->GetOutput()->
+         GetNumberOfPoints(); ++idPixel)
+    {
+      readerIm->GetOutput()->GetPoint(idPixel,pixel);
+
+      pixel[1] = height-1-pixel[1];
+
+      kwiver::vital::vector_3d p;
+
+      double depth = readerIm->GetOutput()->GetPointData()->GetArray("Depths")->
+          GetTuple1(idPixel);
+
+      scaledCam->UnprojectPoint(pixel, depth, &p);
+
+      points->SetPoint(idPixel,p[0],p[1],p[2]);
+    }
+
+    //Generating the PolyData to show
+
+    vtkSmartPointer<vtkPolyData> polyData;
+    vtkNew<vtkGeometryFilter> geometryFilterIm;
+
+    geometryFilterIm->SetInputData(readerIm->GetOutput());
+    geometryFilterIm->Update();
+
+    polyData = geometryFilterIm->GetOutput();
+    polyData->SetPoints(points.Get());
+
+    d->currentDepthmap = polyData.Get();
+    d->currentDepthmap->GetPointData()->SetScalars(d->currentDepthmap->
+                                                   GetPointData()->GetArray("Color"));
+
+    vtkNew<vtkPolyDataMapper> mapper;
+
+    mapper->SetInputData(d->currentDepthmap.Get());
+    mapper->SetColorModeToDirectScalars();
+
+    d->depthmapActor->SetMapper(mapper.Get());
+
+    if(d->depthMapOptions->isPointsChecked())
+    {
+      d->depthmapActor->GetProperty()->SetRepresentationToPoints();
+    }
+    else if(d->depthMapOptions->isSurfacesChecked())
+    {
+      d->depthmapActor->GetProperty()->SetRepresentationToSurface();
+    }
+
+    d->renderer->AddActor(d->depthmapActor.Get());
+
+    d->renderer->AddViewProp(d->depthmapActor.GetPointer());
+
+
+    if(!d->depthMapOptions->isFilterChecked()
+       || !d->depthMapOptions->isFilterPersistChecked())
+    {
+      //initializing the filters
+
+      double bcRange[2], urRange[2];
+
+      d->currentDepthmap->GetPointData()->GetArray("Best Cost Values")->GetRange(bcRange);
+      d->currentDepthmap->GetPointData()->GetArray("Uniqueness Ratios")->GetRange(urRange);
+
+      d->depthMapOptions->initializeFilters(bcRange[0], bcRange[1],
+                                            urRange[0], urRange[1]);
+    }
+    else
+    {
+      updateDepthMapThresholds();
+    }
+
+
+    d->depthmapAlreadLoaded = true;
+  }
+
+  d->currentCam = camera;
+  d->currentVtiPath = vtiPath;
+}
+
+//-----------------------------------------------------------------------------
+void WorldView::enableDepthMap()
```

Comment:
Do we need this? Other entities (e.g. camera image, landmarks) are always enabled, whether or not there is data.

Also, should the depth map visibility state be persisted, as it is for other options? Or are they expensive enough to want them to always default to off?